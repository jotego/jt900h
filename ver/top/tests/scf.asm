    ; set/reset/~zero to carry flag
    main section code
    org 0
    ld a,0xbf    ; common header

    jp c,bad_end
    scf
    jp nc,bad_end
    rcf
    jp c,bad_end

    ld a,0
    add a,a
    jp nz,bad_end
    zcf
    jp c,bad_end
    add a,1
    jp z,bad_end
    zcf
    jp nc,bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end