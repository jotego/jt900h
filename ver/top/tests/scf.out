00000001 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000000
SR = 01f8
